src/codechicken/nei/SubsetWidget.java

```diff
@@ -410,38 +428,80 @@ private static void _setHidden(SubsetTag tag, boolean hidden) {
     }
 
     private static void _setHidden(ItemStack item, boolean hidden) {
-        if(hidden)
-            hiddenItems.add(item);
-        else
-            hiddenItems.remove(item);
+        if (hiddenItemLock.isWriteLockedByCurrentThread()) {
```

Comment:
Note to future: this check is redundant but make it clear that this code is not supposed to be called without write lock held. Overall there isn't much overhead (no synchronization under the hood) so it's a good tradeoff.